src/mpi/coll/src/coll_impl.c

```diff
@@ -256,3 +256,27 @@ void MPIR_Coll_host_buffer_free(void *host_sendbuf, void *host_recvbuf)
     MPL_free(host_sendbuf);
     MPL_free(host_recvbuf);
 }
+
+void MPII_Coll_host_buffer_swap_back(void *host_sendbuf, void *host_recvbuf, void *in_recvbuf,
+                                     MPI_Aint count, MPI_Datatype datatype, MPIR_Request ** request)
```

Comment:
Is it necessary to make `request` INOUT?

Reply:
It's not. I kept the changes to a minimum for copy/paste reasons, but I can update this part. It's a worthwhile improvement.

Reply:
Done.